api/graphql/reservations/reservation_filtersets.py

```diff
@@ -80,3 +110,61 @@ def get_only_with_permission(self, qs, property, value: bool):
             | Q(reservation_unit__unit__service_sectors__in=viewable_service_sectors)
             | Q(user=user)
         ).distinct()
+
+    def get_reservation_unit_name(self, qs, property: str, value: str):
+        language = property[-2:]
+        words = value.split(",")
+        queries = []
+        for word in words:
+            word = word.strip()
+            if language == "en":
+                queries.append(Q(reservation_unit__name_en__istartswith=word))
+            elif language == "sv":
+                queries.append(Q(reservation_unit__name_sv__istartswith=word))
+            else:
+                queries.append(Q(reservation_unit__name_fi__istartswith=word))
+
+        query = reduce(operator.or_, (query for query in queries))
+        return qs.filter(query).distinct()
+
+    def get_unit(self, qs, property, value):
+        if not value:
+            return qs
+
+        return qs.filter(reservation_unit__unit__in=value)
+
+    def get_reservation_unit_type(self, qs, property, value):
+        if not value:
+            return qs
+        return qs.filter(reservation_unit__reservation_unit_type__in=value)
+
+    def get_text_search(serlf, qs, property, value: str):
+        if not value:
+            return qs
+
+        if value.isnumeric():
+            return qs.filter(pk=value)
+
+        queryset = qs.alias(
+            reservee_name=Case(
+                When(
+                    reservee_type=CUSTOMER_TYPES.CUSTOMER_TYPE_BUSINESS,
+                    then=F("reservee_organisation_name"),
+                ),
+                When(
+                    reservee_type=CUSTOMER_TYPES.CUSTOMER_TYPE_NONPROFIT,
+                    then=F("reservee_organisation_name"),
+                ),
+                When(
+                    reservee_type=CUSTOMER_TYPES.CUSTOMER_TYPE_INDIVIDUAL,
+                    then=Concat("reservee_first_name", V(" "), "reservee_last_name"),
+                ),
+                default=V(""),
+                output_field=CharField(),
+            )
+        )
+
+        # Match name
```

Comment:
Does it need to match? Since the code below will apply to partial matches. So should it be exact or is the contains what is wanted?

Reply:
Oops! This comments shouldn't even be here. `icontains` is the one that is expected ☺️ 
I removed the pointless comment.